### ISSUE TITLE
pfstools cannot import pictures in RAW format

### ISSUE THREAD

**Issue by typedfern** [NixOS/nixpkgs#60841]
## Issue description

The package "pfstools" installs a bash script, called "pfsindcraw", for importing camera pictures in raw format.

However, the nix-distributed version fails, because one of the programs it relies on, "pfsinppm", is not compiled (because it requires netpbm being installed).

### Steps to reproduce

1) Install pfstools
2) Try to import a raw camera picture: pfsindcraw camera.raw >camera.pfs

Expected result: The output file "camera.pfs" having several megabytes of PFS data.

Actual result: "pfsinppm program not found. Check if pfstools are compiled with netpbm support."

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.90, NixOS, 18.09.1806.928ff62213b (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(typedfern): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Workaround

I managed to get it to work normally by using tiff instead of pbm as the intermediate format between RAW and PFS; here is the patch:

```diff
diff -pruN a/src/fileformat/pfsindcraw.in b/src/fileformat/pfsindcraw.in
--- a/src/fileformat/pfsindcraw.in      2017-11-13 18:38:27.000000000 +0100
+++ b/src/fileformat/pfsindcraw.in      2018-12-30 17:23:03.837192536 +0100
@@ -28,8 +28,8 @@ if ! which dcraw 2>/dev/null 1>/dev/null
     exit 1;
 fi
 
-if ! which pfsinppm 2>/dev/null 1>/dev/null; then
-    echo >&2 "pfsindcraw: pfsinppm program not found. Check if pfstools are compiled with netpbm support."
+if ! which pfsintiff 2>/dev/null 1>/dev/null; then
+    echo >&2 "pfsindcraw: pfsintiff program not found. Check if pfstools are compiled with TIFF support."
     exit 1;
 fi
 
@@ -56,9 +56,9 @@ fi
 while test "$1"; do
 
       file_pattern="$1"
-
-      dcraw -c -o $COLORSPACE -4 -w "$file_pattern" | pfsinppm - 2> /dev/null | \
-          pfstag --set "FILE_NAME=${file_pattern}" --set "LUMINANCE=RELATIVE"
-      
+      temporary=`mktemp`
+      dcraw -c -o $COLORSPACE -4 -w -T "$file_pattern" >$temporary
+      pfsintiff $temporary 2> /dev/null | pfstag --set "FILE_NAME=${file_pattern}" --set "LUMINANCE=RELATIVE"
+      rm $temporary
       shift
 done
```

**Comment by c0bw3b** [comment]
cc @juliendehos

**Comment by juliendehos** [comment]
Hi @typedfern,
I pushed a fix to my repo (https://github.com/juliendehos/nixpkgs/commit/d4e7deb13b33660dae56b89b8dca8cec6c57a62c) but I have no image for testing it.
Can you try it (`nix-env -iA pfstools -f https://github.com/juliendehos/nixpkgs/tarball/pfstools`) and tell me if works ?

**Comment by typedfern** [comment]
> Hi @typedfern,
> I pushed a fix to my repo ([juliendehos@d4e7deb](https://github.com/juliendehos/nixpkgs/commit/d4e7deb13b33660dae56b89b8dca8cec6c57a62c)) but I have no image for testing it.
> Can you try it (`nix-env -iA pfstools -f https://github.com/juliendehos/nixpkgs/tarball/pfstools`) and tell me if it works ?

Tried it - successfully compiled and installed the (previously missing) pfsinppm command - seems to be fixed.

**Comment by juliendehos** [comment]
Thank you. PR sent.
I added a branch `pfs`, in my repo, that should fix #60841 and #60843. You can use it while the PRs are not merged.

**Comment by typedfern** [comment]
> Thank you. PR sent.
> I added a branch `pfs`, in my repo, that should fix #60841 and #60843. You can use it while the PRs are not merged.

Installed it - both issues appear to be fixed; thank you!
